rm-datasource/src/main/java/io/seata/rm/datasource/exec/DeleteExecutor.java

```diff
@@ -92,13 +92,25 @@ protected TableRecords beforeImage() throws SQLException {
         Statement st = null;
         ResultSet rs = null;
         try {
-            if (paramAppender.isEmpty()) {
+            if (paramAppenders.isEmpty()) {
                 st = statementProxy.getConnection().createStatement();
                 rs = st.executeQuery(selectSQL);
             } else {
-                ps = statementProxy.getConnection().prepareStatement(selectSQL);
-                for (int i = 0; i < paramAppender.size(); i++) {
-                    ps.setObject(i + 1, paramAppender.get(i));
+                if (paramAppenders.size() == 1) {
+                    ps = statementProxy.getConnection().prepareStatement(selectSQL);
+                    for (int i = 0; i < paramAppenders.get(0).size(); i++) {
+                        ps.setObject(i + 1, paramAppenders.get(0).get(i));
+                    }
+                } else {
```

Comment:
When a non-primary key is deleted, paramAppenders.size>1

Reply:
paramAppender represents a set of conditions behind where